language/move-vm/runtime/src/runtime.rs

```diff
@@ -432,18 +435,41 @@ impl VMRuntime {
 
         let mut serialized_mut_ref_outputs = Vec::new();
         for (idx, ty) in mut_ref_inputs {
-            let val = dummy_locals.move_loc(idx).unwrap();
+            let val = match dummy_locals.move_loc(idx) {
+                Ok(v) => v,
+                Err(e) => return Err(e.finish(Location::Undefined)),
+            };
             let layout = self.loader.type_to_type_layout(&ty).map_err(|_err| {
                 PartialVMError::new(StatusCode::INTERNAL_TYPE_ERROR)
                     .with_message("cannot be called with non-serializable return type".to_string())
                     .finish(Location::Undefined)
             })?;
-            let val_bytes = val.simple_serialize(&layout).unwrap();
-            serialized_mut_ref_outputs.push(val_bytes)
+            match val.simple_serialize(&layout) {
+                Some(bytes) => serialized_mut_ref_outputs.push(bytes),
+                None => return Err(PartialVMError::new(StatusCode::INTERNAL_TYPE_ERROR)
+                                .with_message("failed to serialize mutable ref values".to_string())
+                                .finish(Location::Undefined)),
+            };
         }
 
         Ok((serialized_return_vals, serialized_mut_ref_outputs))
     }
+    
+    fn arg_to_actual(&self, idx: usize, arg: Vec<u8>, arg_t: &Box<Type>, locals: &mut Locals, actuals: &mut Vec<Value>) 
```

Comment:
> arg_to_actual() could maybe have a clearer name.

Maybe:
```
fn borrow_arg(&self, idx: usize, arg: Vec<u8>, arg_t: &Box<Type>, locals: &mut Locals) -> Value
```
 , change the function to return the result of `locals.borrow_loc(idx)`, and do the `actuals.push` at the call sites?

Reply:
that sounds fine - mostly I wanted to group all of these just because they're immediately repeated.  Would this make `borrow_arg` more widely useful ?

Reply:
Yes, I think this would help! And the rename is probably a good idea because "arg" and "actual" are synonyms (e.g., like "parameter" and "formal").

Reply:
changed !